application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -39,8 +45,54 @@ export class State extends AdvancedState {
     }
 
     public onClick(event: MouseEvent) {
+        const position: number = this._calculatePosition(event);
+        position >= 0 && this._session.cursor.select(position, Owner.Chart);
+    }
+
+    public onMouseMove(event: MouseEvent) {
+        const rect: DOMRect = this._element.getBoundingClientRect();
+        const offsetX = event.offsetX;
+        if (offsetX > 0) {
+            this.crosshairLeft = offsetX;
+            this.showLeftTooltip = offsetX <= rect.width / 2;
+        }
+        if (event.offsetY > 0) {
+            this.tooltipTop = rect.height / 2;
+            const position: number = this._calculatePosition(event);
```

Comment:
and inside `this._calculatePosition` one more call of `getBoundingClientRect`.
Be careful with this method it should be used as soon as it's really needed. As usual, we can guess a moment when the size could be changed and at this moment we can measure the size.